modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DeliverableAndesMetadata.java

```diff
@@ -0,0 +1,544 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.kernel;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.andes.kernel.slot.Slot;
+import org.wso2.andes.subscription.LocalSubscription;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class represents the message metadata and all the delivery aspects of it to the subscribers (outbound path).
+ * The lifecycle of the message is maintained here itself.
+ */
+public class DeliverableAndesMetadata extends AndesMessageMetadata{
+
+    /**
+     * Map to keep message status and delivery information of this message to vivid channels
+     */
+    private Map<UUID, ChannelInformation> channelDeliveryInfo;
+    /**
+     * State transition of the message
+     */
+    private List<MessageStatus> messageStatus;
+    /**
+     * Parent slot of message.
+     */
+    private Slot slot;
+
+    /**
+     * Time stamp message is read from the store
+     */
+    private long timeMessageIsRead;
```

Comment:
Shall we make this field final? We don't want this to change after creation of the object right?